components/kyma-environment-broker/internal/process/upgrade_kyma/connectivity_bind_test.go

```diff
@@ -0,0 +1,101 @@
+		TokenServiceUrl:                 "tokenserviceurl",
+		TokenServiceUrlPattern:          "tokenserviceurlpattern",
+		TokenServiceUrlPatternTenantKey: "tokenserviceurlpatterntenantkey",
+		Xsappname:                       "xsappname",
+	}
+
+	// when
+	encryptedOverrides, err := provisioning.EncryptConnectivityOverrides(secretKey, &givenOverrides)
+	assert.NoError(t, err)
+	decryptedOverrides, err := provisioning.DecryptConnectivityOverrides(secretKey, encryptedOverrides)
+	assert.NoError(t, err)
+
+	// then
+	assert.Equal(t, givenOverrides, *decryptedOverrides)
+}
+
+func TestConnectivityGetCredentials(t *testing.T) {
+	// given
+	binding := servicemanager.Binding{}
+
+	// when
+	err := json.Unmarshal([]byte(connectivityServiceKey), &binding.Credentials)
+	assert.NoError(t, err)
+	assert.NotNil(t, binding.Credentials)
+
+	// then
+	connOverrides, err := provisioning.GetConnectivityCredentials(binding)
+	assert.NoError(t, err)
+	assert.NotNil(t, connOverrides)
+	assert.Equal(t, "clientid314159265359", connOverrides.ClientId)
+	assert.Equal(t, "Y2xpZW50c2VjcmV0Cg==", connOverrides.ClientSecret)
+	assert.Equal(t, "/api/v1/CAs", connOverrides.ConnectivityService.CAsPath)
+	assert.Equal(t, "/api/v1/CAs/signing", connOverrides.ConnectivityService.CAsSigningPath)
+	assert.Equal(t, "/api/v1/CAs/signing", connOverrides.ConnectivityService.ApiPath)
+	assert.Equal(t, "/api/v1/tunnel", connOverrides.ConnectivityService.TunnelPath)
+	assert.Equal(t, "https://connectivity.company.com", connOverrides.ConnectivityService.Url)
+	assert.Equal(t, "db4dc3bd-3cb7-42d4-a6c0-23aa7842cb7d", connOverrides.SubaccountId)
+	assert.Equal(t, "some-subaccount-subdomain", connOverrides.SubaccountSubdomain)
+	assert.Equal(t, "authentication.company.com", connOverrides.TokenServiceDomain)
+	assert.Equal(t, "https://authentication.company.com/oauth/token", connOverrides.TokenServiceUrl)
+	assert.Equal(t, "https://{tenant}.authentication.company.com/oauth/token", connOverrides.TokenServiceUrlPattern)
+	assert.Equal(t, "subaccount_subdomain", connOverrides.TokenServiceUrlPatternTenantKey)
+	assert.Equal(t, "xsappname314159265359", connOverrides.Xsappname)
+}
+
+const connectivityServiceKey = `
+	{
+		"clientid" : "clientid314159265359",
+		"clientsecret": "Y2xpZW50c2VjcmV0Cg==",
+		"connectivity_service":
+			{
+				"CAs_path": "/api/v1/CAs",
+				"CAs_signing_path": "/api/v1/CAs/signing",
+				"api_path": "/api/v1/CAs/signing",
+				"tunnel_path": "/api/v1/tunnel",
+				"url": "https://connectivity.company.com"
+			},
+		"subaccount_id": "db4dc3bd-3cb7-42d4-a6c0-23aa7842cb7d",
+		"subaccount_subdomain": "some-subaccount-subdomain",
+		"token_service_domain": "authentication.company.com",
+		"token_service_url": "https://authentication.company.com/oauth/token",
+		"token_service_url_pattern": "https://{tenant}.authentication.company.com/oauth/token",
+		"token_service_url_pattern_tenant_key": "subaccount_subdomain",
+		"xsappname": "xsappname314159265359"
+	}
+`
```

Comment:
Similar to my other comment. The difference is that now I know, not to expect the config at the top 😅